hydra-main/src/main/java/com/addthis/hydra/job/web/resources/JobsResource.java

```diff
@@ -1279,7 +1281,26 @@ private static Response buildServerError(Exception exception) {
         if (message == null) {
             message = exception.toString();
         }
-        return Response.serverError().entity(message).build();
+
+        String stack = "";
+        for (StackTraceElement s : exception.getStackTrace()) {
+            stack += s + "\n";
+        }
+
```

Comment:
Yuesong's approach is best.  Very minor comment on building the string though.  Best practice is to use a StringBuilder rather than building then using String +=.  Java strings are immutable so each concatenation creates a new String.  JVM has reduced penalty for this significantly recently but still beneficial to use StringBuilder approach.